src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -203,17 +194,20 @@ final case class JwtUserInfo(sub: IdentityConcentratorId, exp: Long)
 
 object JwtAuthorizationHeader extends LazyLogging {
   /**
-    * Pattern matcher for JWT authorization header.
+    * Pattern matcher for JWT authorization header. Expected cases:
+    * - not a jwt or bearer token: return None
+    * - is a jwt but missing required fields: return Some(Failure, OAuth2BearerToken)
+    * - is a jwt that is parsable to JwtUserInfo: return Some(Success(JwtUserInfo), OAuth2BearerToken)
     *
     * @param authorizationHeader unprocessed value of the Authorization http header
     * @return None if the header is not a bearer token or is not a parsable JWT.
-    * Otherwise it will extract the parsed json of the JWT payload and an OAuth2BearerToken.
+    * Otherwise it will Try to parse the JWT payload to a JwtUserInfo and an OAuth2BearerToken.
     */
-  def unapply(authorizationHeader: String): Option[(JsObject, OAuth2BearerToken)] = {
+  def unapply(authorizationHeader: String): Option[(Try[JwtUserInfo], OAuth2BearerToken)] = {
```

Comment:
Thanks, this is easier for me to follow.
optional consideration: make a case/sum class (whatever you call it) for the three distinct states: the None state, the regrets jwtuserinfo parse case (we never care what the OAuth2BearerToken is if we have regrets about parsing the JwtUserInfo in handleAuthHeaders?), and the tuple (JwtUserInfo, OAuth2BearerToken). I think I draw the line at triple nesting (option, tuple, try), but maybe I'm too faint of heart.